src/modules/uxrce_dds_client/uxrce_dds_client.cpp

```diff
@@ -483,6 +497,87 @@ void UxrceddsClient::handleMessageFormatRequest()
 	}
 }
 
+void UxrceddsClient::checkConnectivity(uxrSession *session)
+{
+	// Reset TX zero counter, when data is sent
+	if (_last_payload_tx_rate > 0) {
+		_num_tx_rate_zero = 0;
+	}
+
+	// Reset RX zero counter, when data is received
+	if (_last_payload_rx_rate > 0) {
+		_num_rx_rate_zero = 0;
+	}
+
+	const hrt_abstime now = hrt_absolute_time();
+
+	// Start ping and tx/rx rate monitoring, unless we're actively sending & receiving payloads successfully
+	if ((_last_payload_tx_rate > 0) && (_last_payload_rx_rate > 0)) {
+		_connected = true;
+		_num_pings_missed = 0;
+		_last_ping = now;
+
+	} else {
+		if (hrt_elapsed_time(&_last_ping) > 1_s) {
+			// Check payload tx rate
+			if (_last_payload_tx_rate == 0) {
+				_num_tx_rate_zero++;
+			}
+
+			// Check payload rx rate
+			if (_last_payload_rx_rate == 0) {
+				_num_rx_rate_zero++;
+			}
+
+			// Check ping
+			_last_ping = now;
+
+			if (_had_ping_reply) {
+				_num_pings_missed = 0;
+
+			} else {
+				++_num_pings_missed;
+			}
+
+			int timeout_ms = 1'000; // 1 second
+			uint8_t attempts = 1;
+			uxr_ping_agent_session(session, timeout_ms, attempts);
+
+			_had_ping_reply = false;
+		}
+
+		if (_num_pings_missed >= 3) {
+			PX4_ERR("No ping response, disconnecting");
+			_connected = false;
+		}
+
+		int32_t tx_timeout = _param_uxrce_dds_tx_to.get();
+		int32_t rx_timeout = _param_uxrce_dds_rx_to.get();
+
+		if (tx_timeout > 0 && _num_tx_rate_zero >= tx_timeout) {
+			PX4_ERR("Payload TX rate zero for too long, disconnecting");
+			_connected = false;
+		}
+
+		if (rx_timeout > 0 && _num_rx_rate_zero >= rx_timeout) {
```

Comment:
Should be
```suggestion
		if (rx_timeout >= 0 && _num_rx_rate_zero >= rx_timeout) {
```
according to the docs, but that makes no sense, so the docs should say "larger than zero"

Reply:
Solved via better doc